src/test/java/com/wikia/webdriver/common/testnglisteners/InvokeMethodAdapter.java

```diff
@@ -33,6 +37,27 @@ public void afterInvocation(IInvokedMethod method, ITestResult result) {
 
   @Override
   public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
-    // TODO Auto-generated method stub
+    if (method.isTestMethod()) {
+      if (isTestExcludedFromEnv(method.getTestMethod().getConstructorOrMethod().getMethod())) {
+        throw new SkipException("Test can't be run on " + Configuration.getEnv() + " environment");
+
+      }
+    }
+  }
+
+  /**
+   * Returns true if test is excluded from running on current test environment
+   */
+  private boolean isTestExcludedFromEnv(Method method) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "isTestExcludedFromEnv" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)